File: ClientApp/src/app/models/DogDto.ts

import { DTOBaseClass } from "./DTOBaseClass";
  
    
import {PersonDto} from "./PersonDto";

import {AddressDto} from "./AddressDto";



 
export class DogDto extends DTOBaseClass{ 


public static AddChildProperty_DogDto_Ownerperson : string= "ownerperson";

public static AddChildProperty_DogDto_Addresses : string= "addresses";



id: string;  

dogname: string;  

ownerpersonid: string;  

ownerperson: PersonDto;  

addresses: AddressDto[];  

}

File: DTO/DogDto.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace normalizrTestProject.DbModels.Dto
{  
    public class DogDto
    {
        public Guid Id { get; set; }

        public string Dogname { get; set; }

        public Guid? Ownerpersonid { get; set; }

        public PersonDto Ownerperson { get; set; }

        public List<AddressDto> Addresses { get; set; }
    }
}